Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm

```diff
@@ -1874,6 +1878,93 @@ static _WKWebExtensionContextError toAPI(WebExtensionContext::Error error)
     }
 }
 
+// MARK: webRequest
+
+bool WebExtensionContext::hasPermissionToSendWebRequestEvent(WebExtensionTab *tab, const URL& resourceURL, const ResourceLoadInfo& loadInfo)
+{
+    if (!tab)
+        return false;
+
+    if (!hasPermission(_WKWebExtensionPermissionWebRequest, tab))
+        return false;
+
+    auto tabURL = tab->url();
+    if (tabURL.isValid() && !hasPermission(tabURL, tab))
+        return false;
+
+    if (resourceURL.isValid() && !hasPermission(resourceURL, tab))
+        return false;
+
+    const URL& resourceLoadURL = loadInfo.originalURL;
+    if (resourceLoadURL.isValid() && !hasPermission(resourceLoadURL, tab))
```

Comment:
Ditto for `isValid()`.

Reply:
Since an empty URL would lead to `hasPermission` returning false, I think I want to keep this check.